File: repo/frontend/src/types/queue.ts
export interface DocumentQueueItem {
  document_id: string
  candidate_id: string
  document_type: string
  current_status: string
  submitted_at: string
  requirement_code: string | null
}

export interface PaymentQueueItem {
  order_id: string
  candidate_id: string
  item_name: string
  amount: string
  payment_method: string
  reference_number: string | null
  submitted_at: string
}

export interface OrderQueueItem {
  order_id: string
  candidate_id: string
  item_name: string
  status: string
  agreed_price: string | null
  updated_at: string
}

export interface ExceptionQueueItem {
  exception_id: string
  candidate_id: string
  anomaly_type: string | null
  status: string
  current_stage: string
  submitted_at: string
}

export interface AfterSalesQueueItem {
  request_id: string
  order_id: string
  candidate_id: string
  request_type: string
  status: string
  window_expires_at: string
  created_at: string
}

File: repo/backend/src/schemas/queue.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel

from ..domain.enums import DocumentStatus, ExceptionStatus, OrderStatus, ReviewStage


class DocumentQueueItem(BaseModel):
    document_id: uuid.UUID
    candidate_id: uuid.UUID
    document_type: str
    current_status: DocumentStatus
    requirement_code: str | None
    updated_at: datetime


class PaymentQueueItem(BaseModel):
    order_id: uuid.UUID
    candidate_id: uuid.UUID
    item_name: str
    amount: Decimal
    payment_method: str
    reference_number: str | None
    submitted_at: datetime


class OrderQueueItem(BaseModel):
    order_id: uuid.UUID
    candidate_id: uuid.UUID
    item_name: str
    status: OrderStatus
    agreed_price: Decimal | None
    updated_at: datetime


class ExceptionQueueItem(BaseModel):
    exception_id: uuid.UUID
    candidate_id: uuid.UUID
    status: ExceptionStatus
    current_stage: ReviewStage
    submitted_at: datetime | None
    created_at: datetime


class AfterSalesQueueItem(BaseModel):
    request_id: uuid.UUID
    order_id: uuid.UUID
    candidate_id: uuid.UUID
    request_type: str
    status: str
    window_expires_at: datetime
    created_at: datetime
